File: backend/models/Curso.py
from __future__ import annotations
from pydantic import BaseModel


class Curso(BaseModel):
    programa: str
    curso: str
    creditos: int
    atributos: list[str] = []
    descripcion: str
    secciones: list[Seccion] = []
  

    def __hash__(self):
        return hash((self.programa,self.curso))
    

# importaciones inferiores para evitar dependencias circulares
from models.Seccion import Seccion


Curso.model_rebuild()

File: src/models/Curso.ts
import Seccion from './Seccion';

export default  class Curso {
  programa: string;
  curso: string;
  creditos: number;
  atributos: string[] = [];
  descripcion: string;
  secciones: Seccion[] = [];

  constructor(programa: string, curso: string, creditos: number, atributos: string[] = [], descripcion: string) {
    this.programa = programa;
    this.curso = curso;
    this.creditos = creditos;
    this.atributos = atributos;
    this.descripcion = descripcion;
  }
}
